### ISSUE TITLE
VisualProspecting still crashing after 2.1.2.1 update

### ISSUE THREAD

**Issue by MadCrafter** [GTNewHorizons/GT-New-Horizons-Modpack#9271]
### Your GTNH Discord Username

Mad__Crafter#2576

### Your Pack Version

2.1.2.1

### Your Server

private server

### Type of Server

Vanilla Forge

### Your Actions

I have tried to run the latest VisualProspecting version (1.0.25b) on my private server. The previous version didn't work and resulted in a crash. That same issue persists in this new version as well. I tried allocating as much RAM as possible to my server, I reduced the max caching usage in the config but it's still printing the same error. What I noticed is that CPU usage spikes up to 100% the second visualprospecting tries to do its thing, with a following error and crash. 

Someone had a similar crash issue a few days ago which was seemingly fixed in the new version, according to discord staff.

Removing visualprospecting fixes the issue.

### Crash Report

[fml-server-latest.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/7774888/fml-server-latest.log)

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the crash still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this crash consistently by following the exact steps I described above.
- [ ] I have asked other people and they confirm they also crash by following the exact steps I described above.

**Comment by bombcar** [comment]
It’s a number reading issue, it is unhappy about something. There may be a chunk with slight corruption or similar but I don’t know how to see what dimension it’s working in. 

Would you be able to upload a zip of the World save somewhere like Google drive?

**Comment by MadCrafter** [comment]
Hm weird for sure. But yeah very possible that something broke in this world, I think it's 2 years old already so it definitely went through a lot of updates lol. Uploaded it on Google Drive: https://drive.google.com/file/d/10fqxRuUGNu_c0AVdvAU1Po1fULlYZju5/view?usp=sharing

Thanks for your help!

**Comment by bombcar** [comment]
```
[16:51:38] [Server thread/ERROR]: Caught exception from visualprospecting
java.lang.NumberFormatException: For input string: "3(1)"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[?:1.8.0_282]
	at java.lang.Integer.parseInt(Integer.java:580) ~[?:1.8.0_282]
	at java.lang.Integer.parseInt(Integer.java:615) ~[?:1.8.0_282]
	at com.sinthoras.visualprospecting.database.cachebuilder.DimensionAnalysis.executeForEachGeneratedOreChunk(DimensionAnalysis.java:95) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at com.sinthoras.visualprospecting.database.cachebuilder.DimensionAnalysis.lambda$processMinecraftWorld$1(DimensionAnalysis.java:43) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183) ~[?:1.8.0_282]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384) ~[?:1.8.0_282]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_282]
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290) ~[?:1.8.0_282]
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731) ~[?:1.8.0_282]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:1.8.0_282]
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401) ~[?:1.8.0_282]
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734) ~[?:1.8.0_282]
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159) ~[?:1.8.0_282]
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173) ~[?:1.8.0_282]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233) ~[?:1.8.0_282]
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485) ~[?:1.8.0_282]
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:650) ~[?:1.8.0_282]
	at com.sinthoras.visualprospecting.database.cachebuilder.DimensionAnalysis.processMinecraftWorld(DimensionAnalysis.java:42) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at com.sinthoras.visualprospecting.database.cachebuilder.WorldAnalysis.cacheVeins(WorldAnalysis.java:27) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at com.sinthoras.visualprospecting.hooks.HooksShared.fmlLifeCycleEvent(HooksShared.java:92) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at com.sinthoras.visualprospecting.hooks.HooksClient.fmlLifeCycleEvent(HooksClient.java:49) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at com.sinthoras.visualprospecting.VPMod.fmlLifeCycle(VPMod.java:48) ~[visualprospecting-1.7.10-1.0.25b.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_282]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_282]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_282]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_282]
	at cpw.mods.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:532) ~[forge-1.7.10-10.13.4.1614-1.7.10-universal.jar:?]
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_282]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_282]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) ~[guava-17.0.jar:?]
	at cpw.mods.fml.common.LoadController.sendEventToModContainer(LoadController.java:212) ~[forge-1.7.10-10.13.4.1614-1.7.10-universal.jar:?]
	at cpw.mods.fml.common.LoadController.propogateStateMessage(LoadController.java:190) ~[forge-1.7.10-10.13.4.1614-1.7.10-universal.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_282]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_282]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_282]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_282]
	at com.google.common.eventbus.EventSubscriber.handleEvent(EventSubscriber.java:74) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.SynchronizedEventSubscriber.handleEvent(SynchronizedEventSubscriber.java:47) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatch(EventBus.java:322) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.dispatchQueuedEvents(EventBus.java:304) ~[guava-17.0.jar:?]
	at com.google.common.eventbus.EventBus.post(EventBus.java:275) ~[guava-17.0.jar:?]
	at cpw.mods.fml.common.LoadController.distributeStateMessage(LoadController.java:119) [LoadController.class:?]
	at cpw.mods.fml.common.Loader.serverStarting(Loader.java:789) [Loader.class:?]
	at cpw.mods.fml.common.FMLCommonHandler.handleServerStarting(FMLCommonHandler.java:278) [FMLCommonHandler.class:?]
	at net.minecraft.server.integrated.IntegratedServer.func_71197_b(IntegratedServer.java:94) [bsx.class:?]
	at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:387) [MinecraftServer.class:?]
	at net.minecraft.server.MinecraftServer$2.run(MinecraftServer.java:685) [?:?]
```

(3)1 but basically the same error, let's see if VP has some debug code

**Comment by bombcar** [comment]
This is why:

<img width="216" alt="image" src="https://user-images.githubusercontent.com/533931/147373870-fee78f60-7577-4bdc-8536-e1e833423699.png">

Those are invalid region filenames - something (somewhere) made those - maybe on a region restore via some backup tool?

Here's a JAR to try (unzip to get the jar out and replace existing):
[visualprospecting-1.7.10-1.0.25b.dirty.jar.zip](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/7775229/visualprospecting-1.7.10-1.0.25b.dirty.jar.zip)

**Comment by bombcar** [comment]
As a note for me on MultiMC when VP dies in this, it somehow keeps a thread open and I have to kill the Java process.

**Comment by bombcar** [comment]
```
[17:15:32] [ForkJoinPool.commonPool-worker-6/WARN]: [VisualProspecting] Invalid region file found! r.1.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-4/WARN]: [VisualProspecting] Invalid region file found! r.-3.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-3/WARN]: [VisualProspecting] Invalid region file found! r.-2.1(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-0/WARN]: [VisualProspecting] Invalid region file found! r.-2.2(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-4/WARN]: [VisualProspecting] Invalid region file found! r.-2.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-4/WARN]: [VisualProspecting] Invalid region file found! r.-1.3(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-6/WARN]: [VisualProspecting] Invalid region file found! r.0.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-6/WARN]: [VisualProspecting] Invalid region file found! r.-1.2(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-1/WARN]: [VisualProspecting] Invalid region file found! r.-1.1(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-7/WARN]: [VisualProspecting] Invalid region file found! r.-1.0(1).mca continuing
[17:15:32] [Server thread/INFO]: [VisualProspecting] Processing dimension with id 85 with fast scanning.
[17:15:32] [Server thread/INFO]: [VisualProspecting] Processing dimension with id 49 with fast scanning.
[17:15:32] [Server thread/INFO]: [VisualProspecting] Processing dimension with id 82 with fast scanning.
[17:15:32] [Server thread/INFO]: [VisualProspecting] Processing dimension with id -1 with slow (safe) scanning.
[17:15:32] [ForkJoinPool.commonPool-worker-4/WARN]: [VisualProspecting] Invalid region file found! r.0.1(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-7/WARN]: [VisualProspecting] Invalid region file found! r.-1.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-1/WARN]: [VisualProspecting] Invalid region file found! r.-2.1(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-1/WARN]: [VisualProspecting] Invalid region file found! r.0.0(1).mca continuing
[17:15:32] [ForkJoinPool.commonPool-worker-1/WARN]: [VisualProspecting] Invalid region file found! r.-1.1(1).mca continuing
```

seems to be loading

**Comment by MadCrafter** [comment]
Yup, I just tried it out and it worked flawlessly for me. Very nice. :D This is honestly one of the coolest additions in this update and I'm happy it finally works. Yea and those duplicate region files have been there for the longest time. Probably coming from a backup I messed up. Just for confirmation though: Is it safe for me to delete all these duplicate region names in brackets? Or could that mess up my world save somehow?

Thanks, appreciate the effort.
